File: nestjs-api/src/types/dto/addplayer.dto.ts
import { AutoMap } from '@automapper/classes';

export class AddplayerDto {
  @AutoMap()
  fullName: string;

  @AutoMap()
  hightInCm: number;

  @AutoMap()
  weightInKg: number;

  @AutoMap()
  club: string;

  @AutoMap()
  position: string;

  @AutoMap()
  age: number;

  @AutoMap()
  nationId: number;
}

File: dotnet-api/Models/DTO/AddPlayerDTO.cs
using Microsoft.AspNetCore.Http.HttpResults;
using System.ComponentModel.DataAnnotations;

namespace dotnet_api.Models.DTO
{
    public class AddPlayerDTO
    {
        public string FullName { get; set; }
        public int HightInCm { get; set; }
        public int WeightInKg { get; set; }
        public string Club { get; set; }
        public string Position { get; set; }
        public int Age { get; set; }
        public int NationId { get; set; }
    }
}
